packages/protocol/contracts/layer1/based/TaikoInbox.sol

```diff
@@ -183,15 +179,54 @@ abstract contract TaikoInbox is EssentialContract, ITaikoInbox, IProposeBatch, I
                     blobByteOffset: params.blobParams.byteOffset,
                     blobByteSize: params.blobParams.byteSize,
                     gasLimit: config.blockMaxGasLimit,
-                    lastBlockId: lastBatch.lastBlockId + uint64(params.blocks.length),
-                    lastBlockTimestamp: lastBlockTimestamp,
+                    lastBlockId: lastBatch.lastBlockId + uint64(nBlocks),
+                    lastBlockTimestamp: _validateBatchParams(
+                        params,
+                        config.maxAnchorHeightOffset,
+                        config.maxSignalsToReceive,
+                        config.maxBlocksPerBatch,
+                        lastBatch
+                    ),
                     // Data for the L2 anchor transaction, shared by all blocks in the batch
-                    anchorBlockId: anchorBlockId,
-                    anchorBlockHash: blockhash(anchorBlockId),
+                    anchorBlockIds: new uint64[](nBlocks), // to be initialised later
+                    anchorBlockHashes: new bytes32[](nBlocks), // to be initialised
+                        // later
                     baseFeeConfig: config.baseFeeConfig
                 });
 
-                require(info_.anchorBlockHash != 0, ZeroAnchorBlockHash());
+                for (uint256 i; i < nBlocks; ++i) {
+                    uint64 anchorBlockId = params.blocks[i].anchorBlockId;
+                    if (anchorBlockId != 0) {
+                        if (lastAnchorBlockId == 0) {
+                            // This is the first non zero anchor block id in the batch.
+                            require(
+                                anchorBlockId + config.maxAnchorHeightOffset >= block.number,
+                                AnchorIdTooLarge()
```

Comment:
Shouldn't this be:
```suggestion
                                AnchorIdTooSmall()
```

We are checking that anchor id isn't too far in the past, i.e., too small.

Note that `AnchorIdTooSmall` doesn't exist in current ver so this won't compile.